Add LBH with Anti-Grief, consider replacement of old method.

MalefactorIX opened the pull request:
The anti-grief shown works against both large AND recursive damage sources without filling up the list with each individual hit. This is, obviously, slower and will more likely drops hits during high latency than the slim or directional variants BUT it should be very rare if not virtually impossible for it to stack-heap under stressful conditions. However since damage is still updated at the time of impact and instead of waiting on the timer event, it should overall perform better in a majority of situations.

MalefactorIX committed:
Optimization Pass

Removed code that was used in the Directional version which now serves no function in slim

MalefactorIX committed:
Create LBH-AG v1.2.3.lsl

The anti-grief shown works against both large AND recursive damage sources without filling up the list with each individual hit. This is, obviously, slower and will more likely drops hits during high latency BUT it should be very rare if not virtually impossible for it to stack-heap under stressful conditions. However since damage is still updated at the time of impact and instead of waiting on the timer event, it should overall perform better in a majority of situations.

MalefactorIX committed:
Fixed missing " and deprecate variables

MalefactorIX committed:
Reverted unintended edit

MalefactorIX committed:
Reeeee healing

MalefactorIX committed:
Fixed conflict with scripts that look for LBA.V.L

MalefactorIX committed:
Update and rename LBHD, Directional Armor v1.2.4.lsl to LBHD, Directional Armor v1.2.5.lsl

MalefactorIX committed:
v1.3 'Fixed-Location' Update

The intent is to grab the source's location with llDetectedPos during the collision event and using that as a reference. If no such reference exists, the system will just default to the current method. 

The goal is to clamp how the list is generated so it doesn't cause the system to stack-heap when exposed to many damage sources at a time. The eldest entries in the list will be dropped once it fills.

MalefactorIX committed:
Create LBHD+HD, v1.3.0.lsl

MalefactorIX committed:
Rename LBHD+HD, v1.3.0.lsl to LBHD+AG, v1.3.0.lsl

MalefactorIX committed:
Fixed desc conflict with certain weapons

MalefactorIX committed:
Fixed desc conflict for certain weapons

MalefactorIX committed:
Minor tweak to previous change to keep in theme

MalefactorIX committed:
Rename LBH-AG v1.2.3.lsl to LBH-AG v1.2.4.lsl

MalefactorIX committed:
Streamlined 'fixed-location' method

Now calculates and stores the modifier instead of the position. Removed a bunch of commands and otherwise unused code.

MalefactorIX committed:
Updated 'fixed-location' method

HadetTheUndying commented:
As per Criss' report in Discord it appears that Damage Messages from things that don't collide with the Armored object are blocked. For Example

```
[11:57] [Chaos] Voss Pattern Lascannon v.1.17 : Delivering 25 AT to - Object
[11:57] Object: Damage Blocked by Armor
[11:57] Object: Attack was stopped by armor.
```
I'll try to Lint the code in a bit, but when it was initially reported I thought it was an issue with the merged 1.2.3. So I logged in to verify.

MalefactorIX committed:
Oops

MalefactorIX committed:
Oops

MalefactorIX committed:
Rename LBHD+AG, v1.3.1.lsl to LBHD+AG, v1.3.2.lsl

MalefactorIX committed:
Rename LBHD, Directional Armor v1.3.1.lsl to LBHD, Directional Armor v1.3.2.lsl

MalefactorIX committed:
Version fix

HadetTheUndying commented:
I'll do a bit more testing on your fork on Saturday, and probably Merge on Monday.

I don't really see any issues, and I agree it might be time to consider reworking the existing Anti-Grief system in the core LBA.

MalefactorIX committed:
Fix for raycast modifier triggers

MalefactorIX committed:
Raycast fix, more data for owner when damaged

MalefactorIX committed:
Forced LoS on RC Fix

MalefactorIX committed:
Rename LBHD+AG, v1.3.4.lsl to LBHD+AG, v1.3.5.lsl

MalefactorIX committed:
Forced LoS on RC Fix

MalefactorIX committed:
woops

MalefactorIX committed:
Text cleanup

MalefactorIX committed:
Increased information given to owner when damaged

MalefactorIX committed:
Rename LBH Slim v1.2.4.lsl to LBH/LBH Slim v1.2.4.lsl

MalefactorIX committed:
Rename LBH-AG v1.2.4.lsl to LBH/LBH-AG v1.2.4.lsl

MalefactorIX committed:
Rename LBHD+AG, v1.3.5.lsl to LBH/LBHD+AG, v1.3.5.lsl

MalefactorIX committed:
Rename LBHD, Directional Armor v1.3.5.lsl to LBH/LBHD, Directional Armor v1.3.5.lsl

MalefactorIX committed:
Rename LBHD+AG, v1.3.5.lsl to LBHD+AG, Directional Armorv1.3.5.lsl

MalefactorIX committed:
Rename LBHD+AG, Directional Armorv1.3.5.lsl to LBHD+AG, Directional Armor v1.3.5.lsl

MalefactorIX committed:
Merged LBAPOS and COLLISIONMOD

Fixed missing parameter in damage function

MalefactorIX committed:
Merged LBAPOS and COLLISIONMOD

MalefactorIX committed:
Rename LBHD+AG, Directional Armor v1.3.5.lsl to LBHD+AG, Directional Armor v1.3.6.lsl

MalefactorIX committed:
Create LBHD-Component, v1.3.6.lsl

MalefactorIX committed:
Create README.md

MalefactorIX committed:
Update README.md

MalefactorIX committed:
Update README.md

MalefactorIX committed:
Update LBHD+AG, Directional Armor v1.3.6.lsl

MalefactorIX committed:
Update LBHD, Directional Armor v1.3.6.lsl

MalefactorIX committed:
Update README.md

MalefactorIX committed:
Update README.md

MalefactorIX committed:
Update README.md

MalefactorIX committed:
Incline fix

Fixed inclines/declines causing modifiers to be miscalculated. The first vector in the RotBetween did not have its z-axis removed.

MalefactorIX committed:
Incline fix

MalefactorIX committed:
Incline fix

HadetTheUndying commented:
I'm gonna be testing this over the weekend. If you're satisfied that it's stable, I'll merge after testing.

MalefactorIX committed:
Oops

Fixed missing condition

MalefactorIX committed:
Oops

MalefactorIX committed:
No longer supported. LBA or GTFO

MalefactorIX committed:
Updated note with current use-case

MalefactorIX committed:
Updated note with current use-case

MalefactorIX committed:
Resolved slight disagreement

MalefactorIX committed:
v1.4 Threshold Update

Top and Bottom thresholds have been separated for more versatile configurations.

MalefactorIX committed:
v1.4 Threshold Update

MalefactorIX committed:
Rename LBHD-Component, v1.3.7.lsl to LBHD-Component, v1.4.lsl

MalefactorIX committed:
v1.4 Threshold Update

MalefactorIX committed:
Bye Nerd

MalefactorIX committed:
Clarified legacy support

HadetTheUndying commented:
Are you ready for me to test this this week or do you think you're gonna make a ton more commits? Sorry it took me so long to get around to it dude.

MalefactorIX committed:
v1.4.1 Update

- Adjusted detection for some LBA sources rezzed by a non-attached object

MalefactorIX committed:
Update and rename LBHD, Directional Armor v1.4.lsl to LBHD, Directional Armor v1.4.1.lsl

MalefactorIX committed:
Update and rename LBHD+AG, Directional Armor v1.4.lsl to LBHD+AG, Directional Armor v1.4.1.lsl

MalefactorIX committed:
Fixed incorrect typecast

MalefactorIX committed:
Rotational fix for top/bottom modifiers(TEST THIS)

MalefactorIX committed:
Update LBHD-Component, v1.4.3.lsl

MalefactorIX committed:
Update and rename LBHD+AG, Directional Armor v1.4.1.lsl to LBHD+AG, Directional Armor v1.4.2.lsl

MalefactorIX committed:
Update and rename LBHD, Directional Armor v1.4.1.lsl to LBHD, Directional Armor v1.4.2.lsl

MalefactorIX committed:
How did I not catch this

MalefactorIX committed:
Update LBHD+AG, Directional Armor v1.4.3.lsl

MalefactorIX committed:
Fixed healing cap not working

MalefactorIX committed:
Update and rename LBHD-Component, v1.4.3.lsl to LBHD-Component, v1.4.4.lsl

MalefactorIX committed:
Update LBHD, Directional Armor v1.4.3.lsl

MalefactorIX committed:
Update LBHD+AG, Directional Armor v1.4.3.lsl

MalefactorIX committed:
Update and rename LBH-AG v1.2.4.lsl to LBH-AG v1.2.5.lsl

MalefactorIX committed:
Update and rename LBH Slim v1.2.4.lsl to LBH Slim v1.2.5.lsl
